scene/2d/tile_map.h

```diff
@@ -108,62 +157,170 @@ struct TileMapQuadrant {
-		}
+private:
+	// Exposed properties
+	String name;
+	bool enabled = true;
+	Color modulate = Color(1, 1, 1, 1);
+	bool y_sort_enabled = false;
+	int y_sort_origin = 0;
+	int z_index = 0;
+	RID navigation_map;
+	bool uses_world_navigation_map = false;
+
+	// Internal
+	TileMap *tile_map_node;
+	int layer_index_in_tile_map_node = -1;
+	RID canvas_item;
+	bool _rendering_quadrant_order_dirty = false;
+	HashMap<Vector2i, TileMapCell> tile_map;
+	HashMap<Vector2i, TileMapQuadrant> quadrant_map;
+	SelfList<TileMapQuadrant>::List dirty_quadrant_list;
+
+	// Rect cache
+	mutable Rect2 rect_cache;
+	mutable bool rect_cache_dirty = true;
+	mutable Rect2i used_rect_cache;
+	mutable bool used_rect_cache_dirty = true;
+
+	// Quadrants management
+	Vector2i _coords_to_quadrant_coords(const Vector2i &p_coords) const;
+	HashMap<Vector2i, TileMapQuadrant>::Iterator _create_quadrant(const Vector2i &p_qk);
+	void _make_quadrant_dirty(HashMap<Vector2i, TileMapQuadrant>::Iterator Q);
+	void _erase_quadrant(HashMap<Vector2i, TileMapQuadrant>::Iterator Q);
 
-		Vector2i get_base_cell_coords() const {
-			return base_cell_coords;
-		}
+	// Per-system methods.
+	void _rendering_notification(int p_what);
+	void _rendering_update();
+	void _rendering_cleanup();
+	void _rendering_update_dirty_quadrants(SelfList<TileMapQuadrant>::List &r_dirty_quadrant_list);
+	void _rendering_reorder_quadrants(int &r_index);
+	void _rendering_create_quadrant(TileMapQuadrant *p_quadrant);
+	void _rendering_cleanup_quadrant(TileMapQuadrant *p_quadrant);
+	void _rendering_draw_quadrant_debug(TileMapQuadrant *p_quadrant);
 
-		bool is_center_bit() const {
-			return bit == 0;
-		}
+	HashMap<RID, Vector2i> bodies_coords; // Mapping for RID to coords.
+	void _physics_update_dirty_quadrants(SelfList<TileMapQuadrant>::List &r_dirty_quadrant_list);
+	void _physics_cleanup_quadrant(TileMapQuadrant *p_quadrant);
+	void _physics_draw_quadrant_debug(TileMapQuadrant *p_quadrant);
 
-		HashMap<Vector2i, TileSet::CellNeighbor> get_overlapping_coords_and_peering_bits() const;
+	void _navigation_update();
+	void _navigation_cleanup();
+	void _navigation_update_dirty_quadrants(SelfList<TileMapQuadrant>::List &r_dirty_quadrant_list);
+	void _navigation_cleanup_quadrant(TileMapQuadrant *p_quadrant);
+	void _navigation_draw_quadrant_debug(TileMapQuadrant *p_quadrant);
 
-		void set_terrain(int p_terrain) {
-			terrain = p_terrain;
-		}
+	HashSet<Vector2i> instantiated_scenes;
+	void _scenes_update_dirty_quadrants(SelfList<TileMapQuadrant>::List &r_dirty_quadrant_list);
+	void _scenes_cleanup_quadrant(TileMapQuadrant *p_quadrant);
+	void _scenes_draw_quadrant_debug(TileMapQuadrant *p_quadrant);
 
-		int get_terrain() const {
-			return terrain;
-		}
+	// Runtime tile data
+	void _build_runtime_update_tile_data(SelfList<TileMapQuadrant>::List &r_dirty_quadrant_list);
 
-		void set_priority(int p_priority) {
-			priority = p_priority;
-		}
+	// Terrains.
```

Comment:
Either all these labels should end with period or none.
We usually add the period, but the comments here are not sentences, so not sure.